jupyterlab_gitsync/src/components/toolbar_buttons.tsx

```diff
@@ -1,118 +1,61 @@
-import * as React from 'react';
-import { classes, style } from 'typestyle';
-import {
-  toolbarItemClass,
-  toolbarButtonClass,
-  toolbarButtonIconClass,
-} from '../style/toolbar';
+import { style } from 'typestyle';
+import { ToolbarButton } from './toolbar_btn_template';
 
-import { Props } from './panel';
-
-export interface State {
-  state: string;
-  label: string;
-  style: ReturnType<typeof style>;
-}
-
-class ToolbarButton extends React.Component<Props, State> {
-  constructor(
-    props: Props,
-    init_label: string,
-    init_style: ReturnType<typeof style>
-  ) {
-    super(props);
-    this.state = {
-      state: 'Running',
-      label: init_label,
-      style: init_style,
-    };
-  }
-
-  render(): React.ReactElement {
-    return (
-      <div
-        className={classes(
-          'jp-ToolbarButton',
-          'jp-Toolbar-item',
-          toolbarItemClass
-        )}
-      >
-        <button
-          className={classes(
-            'bp3-button',
-            'bp3-minimal',
-            'jp-ToolbarButtonComponent',
-            'minimal',
-            'jp-Button',
-            toolbarButtonClass
-          )}
-          title={this.state.label}
-          onClick={this._onClick}
-        >
-          <span className={classes('bp3-button-text')}>
-            <span
-              className={classes(
-                'jp-ToolbarButtonComponent-icon',
-                toolbarButtonIconClass,
-                this.state.style
-              )}
-            >
-              {' '}
-            </span>
-          </span>
-        </button>
-      </div>
-    );
-  }
-
-  protected _onClick = (): void => {
-    console.log('hi');
-  };
-}
+// TO DO (ashleyswang): Change to non-inheritance implementation of React Component
 
 export class ControlButton extends ToolbarButton {
   constructor(props) {
     super(
       props,
       'Stop Auto Sync',
-      style({ backgroundImage: 'var(--jp-icon-stop)' })
+      style({ backgroundImage: 'var(--jp-icon-stop)' }), 
+      { running: true },
     );
 
     this.props.service.start();
+    this._addListeners();
   }
 
   protected _onClick = (): void => {
-    if (this.props.service.isRunning) {
-      this._stopSync();
+    if (this.props.service.running) {
+      this.props.service.stop();
```

Comment:
Seems to me like this toggles the auto sync feature, so if that's the case, wouldn't the state also need to be set here?

Reply:
When we call `service.stop()/.start()`, the service will emit a signal indicating whether or not its state has changed. This is done because there are cases in which I believe the sync should not be able to start even if it is currently stopped, such as if it is in a blocked state due to there being unresolved conflicts in a file. 

Instead, we connect the button to an event listener so that if the service state change is successful, then we are allowed to change the state of the button to match the service. By making the button state depend on the service state, we can ensure that they are always synced and that the button always shows the correct state of the service.

Reply:
Ah I see makes sense.